include/qpid/dispatch/alloc_pool.h

```diff
@@ -80,7 +80,26 @@ void *qd_alloc(qd_alloc_type_desc_t *desc, qd_alloc_pool_t **tpool);
 /** De-allocate from a thread pool. Use via ALLOC_DECLARE */
 void qd_dealloc(qd_alloc_type_desc_t *desc, qd_alloc_pool_t **tpool, char *p);
 uint32_t qd_alloc_sequence(void *p);
-static inline void qd_nullify_safe_ptr(qd_alloc_safe_ptr_t *sp) { sp->ptr = 0; }
+
+// generic safe pointer api for any alloc pool item
+
+#define QD_SAFE_PTR_INIT(p) { .ptr = (void*)(p), .seq = qd_alloc_sequence(p) }
+
+static inline void qd_nullify_safe_ptr(qd_alloc_safe_ptr_t *sp)
```

Comment:
Instead of this function (qd_nullify_safe_ptr), can we just pass in a zero to qd_alloc_set_safe_ptr(). That way when people search for qd_alloc_set_safe_ptr, we can see where the safe pointer is set and unset. I do understand that this PR is not introducing the qd_nullify_safe_ptr() function.

Reply:
I think qd_nullify.. is used in other places outside of Q2.  A change like that probably deserves a separate patch.